TOOLS/lua/autocrop.lua

```diff
@@ -85,14 +101,19 @@ function do_crop()
         if cropdetect_metadata["lavfi.cropdetect.w"]
             and cropdetect_metadata["lavfi.cropdetect.h"]
             and cropdetect_metadata["lavfi.cropdetect.x"]
-            and cropdetect_metadata["lavfi.cropdetect.y"]
-        then
-            mp.command(string.format("vf add @%s:lavfi-crop=w=%s:h=%s:x=%s:y=%s",
-                                     crop_label,
-                                     cropdetect_metadata["lavfi.cropdetect.w"],
-                                     cropdetect_metadata["lavfi.cropdetect.h"],
-                                     cropdetect_metadata["lavfi.cropdetect.x"],
-                                     cropdetect_metadata["lavfi.cropdetect.y"]))
+            and cropdetect_metadata["lavfi.cropdetect.y"] then
+            if tonumber(cropdetect_metadata["lavfi.cropdetect.w"]) > mp.get_property_native("width") * options.crop_max_ratio
+                and tonumber(cropdetect_metadata["lavfi.cropdetect.h"]) > mp.get_property_native("height") * options.crop_max_ratio
+                and (tonumber(cropdetect_metadata["lavfi.cropdetect.x"]) > 0
+                    or tonumber(cropdetect_metadata["lavfi.cropdetect.y"]) > 0)
```

Comment:
I think disallowing 0 offset to be a little heavy handed

Reply:
`if("0")
then
    print("true")
else
    print("false")
end` >> true.
Obviously "vf=@autocrop-crop:lavfi-crop=w=1920:h=1080:x=0:y=0" is invalid

Reply:
ok, fair enough, but we should probably check all four parameters for noop rather than assuming the other two will be what we expect

Reply:
`meta.w >= meta.min_w and meta.h >= meta.min_h` ensure that two are greater than 0

![image](https://user-images.githubusercontent.com/50797982/82864981-54fc9800-9f58-11ea-9f15-d1097b594d58.png)
These two will have negative values when screen full black. (So automatic tasks need to be delayed)